src/compiler/checker.ts

```diff
@@ -12619,6 +12643,13 @@ namespace ts {
                         }
                     }
                     checkKindsOfPropertyMemberOverrides(type, baseType);
+                        
+                    // Check if base type declaration appears before heritage clause to avoid false errors for 
+                    // base type declarations in the extend clause itself
+                    let baseTypeDeclaration = getLeftmostAliasDeclarationFromExpressionOrEnityName(baseTypeNode.expression) || baseType.symbol.declarations[0];
+                    if (!isDefinedBefore(baseTypeDeclaration, baseTypeNode)) {

```

Comment:
Why/ because we get it either as alias in the expression use or base type expression's symbol's declaration which should always be there isn't it?